src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1734,6 +1734,14 @@ def update_app_service_plan(instance, sku=None, number_of_workers=None):
     return instance
 
 
+def show_plan(cmd, resource_group_name, name):
+    client = web_client_factory(cmd.cli_ctx)
+    plan = client.app_service_plans.get(resource_group_name, name)
+    if not plan:
+        raise ResourceNotFoundError("Appservice Plan'{}', is not found on RG '{}'.".format(name, resource_group_name))
```

Comment:
https://github.com/Azure/azure-cli/blob/3f230e856a407043102e99b82d4d22f679111a3c/src/azure-cli-core/azure/cli/core/commands/arm.py#L408-L416

If your server could raise exception with status code 404, ResourceNotFound error would be raised. So why do you do the change now?

Reply:
API is not throwing the exception. But it is returning 404 status code. to handle that I prefer to go with custom exception.

Reply:
@Juliehzl- API is not returning any Json response but it is returning html response. please suggest me the best approach of handling html response in CLI.

Reply:
As you define 404 status code as non-error in [swagger](https://github.com/Azure/azure-rest-api-specs/blob/c803720c6bcfcb0fcf4c97f3463ec33a18f9e55c/specification/web/resource-manager/Microsoft.Web/stable/2021-01-01/AppServicePlans.json#L140-L142) , the generated python SDK will not raise exception as [here](https://github.com/Azure/azure-sdk-for-python/blob/f89776994d2a0292e6b10b9920f8828d5554143a/sdk/appservice/azure-mgmt-web/azure/mgmt/web/v2021_01_01/operations/_app_service_plans_operations.py#L250-L253) and azure CLI cannot catch the exception for resourceNotFound error.

There are two ways to fix:
1. Change swagger definition, which need to remove 404 part in response as [storage](https://github.com/Azure/azure-rest-api-specs/blob/0c057560083bfdfd50202f0e79cfac8ec054d56a/specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/storage.json#L383-L398) and [network](https://github.com/Azure/azure-rest-api-specs/blob/0c057560083bfdfd50202f0e79cfac8ec054d56a/specification/network/resource-manager/Microsoft.Network/stable/2021-02-01/applicationGateway.json#L126-L140).
2. If you don't want to change swagger, you need to catch it in your code. In this way, I would recommend  you check response status code with 404. Currently you will raise resource not error when not plan, which is not accurate enough.

Reply:
@panchagnula your thoughts please.